stack/CO_SYNC.c

```diff
@@ -337,13 +338,24 @@ uint8_t CO_SYNC_process(
 
         /* SYNC producer */
         if(SYNC->isProducer && SYNC->periodTime){
+            uint32_t diff;
             if(SYNC->timer >= SYNC->periodTime){
                 if(++SYNC->counter > SYNC->counterOverflowValue) SYNC->counter = 1;
                 SYNC->timer = 0;
                 ret = 1;
                 SYNC->CANrxToggle = SYNC->CANrxToggle ? false : true;
                 SYNC->CANtxBuff->data[0] = SYNC->counter;
                 CO_CANsend(SYNC->CANdevTx, SYNC->CANtxBuff);
+                diff = 0;
```

Comment:
Just to confirm, does the processing function not need to immediately run when the timer has expired?

Reply:
Actually, I guess if it subscribes to the SYNC itself it would immediately trigger the interrupt anyway and process again, so then the next time it would need to run is the next periodTime. I think I understand now.